### PR TITLE
feat: add C implementation for `math/base/special/cos`

### PR DESCRIPTION
**Pull request by gunjjoshi**
## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for [`math/base/special/cos`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/cos).

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves a part of #649.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

I have added the extended comment from the [`FreeBSD`](https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/s_cos.c?revision=367086&view=markup#l24) source in both the `C` and `javascript` implementations.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/cos/src/main.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+*
+*     | n | sin(x) | cos(x) | tan(x) |
+*     | - | ------ | ------ | ------ |
+*     | 0 |   S    |   C    |    T   |
+*     | 1 |   C    |  -S    |  -1/T  |
+*     | 2 |  -S    |  -C    |    T   |
+*     | 3 |  -C    |   S    |  -1/T  |
+*
+* @param x    input value (in radians)
+* @return    cosine
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/cos/src/main.c]: Missing a space to align with `x` parameter description:

```suggestion
* @return     cosine
```

**Comment by Planeshifter** [comment]
@gunjjoshi Thank you, @gunjjoshi! Will merge shortly.

**Review by Planeshifter — APPROVED** [review]

File: lib/node_modules/@stdlib/math/base/special/cos/include/stdlib/math/base/special/cos.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+* limitations under the License.
+*/
+
+/**
+* Header file containing function declarations.
+*/
+#ifndef STDLIB_MATH_BASE_SPECIAL_COS_H
+#define STDLIB_MATH_BASE_SPECIAL_COS_H
+
+#include <stdint.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cos/include/stdlib/math/base/special/cos.h:25]: This include is not necessary.

File: lib/node_modules/@stdlib/math/base/special/cos/src/main.c [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+#include "stdlib/math/base/special/cos.h"
+#include "stdlib/number/float64/base/get_high_word.h"
+#include "stdlib/math/base/special/kernel_cos.h"
+#include "stdlib/math/base/special/kernel_sin.h"
+#include "stdlib/math/base/special/rempio2.h"
+#include <stdint.h>
+
+// High word absolute value mask: 0x7fffffff => 01111111111111111111111111111111
+static const int32_t HIGH_WORD_ABS_MASK = 0x7fffffff;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cos/src/main.c:41]: I believe we have a constant package for this.

File: lib/node_modules/@stdlib/math/base/special/cos/src/main.c [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+// High word of π/4: 0x3fe921fb => 00111111111010010010000111111011
+static const int32_t HIGH_WORD_PIO4 = 0x3fe921fb;
+
+// High word of 2^-27: 0x3e400000 => 00111110010000000000000000000000
+static const int32_t HIGH_WORD_TWO_NEG_27 = 0x3e400000;
+
+// High word exponent mask: 0x7ff00000 => 01111111111100000000000000000000
+static const int32_t HIGH_WORD_EXPONENT_MASK = 0x7ff00000;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cos/src/main.c:50]: Same comment. We have a pkg for this, IIRC.

**Comment by gunjjoshi** [comment]
@kgryte I've made a new PR incorporating the changes at #2355.

**Pull request merged**